RestaurantReservation.Db/RestaurantReservationDbContext.cs

```diff
@@ -0,0 +1,214 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.Models;
+
+namespace RestaurantReservation.Db;
+public class RestaurantReservationDbContext : DbContext
+{
+    public DbSet<Customer> Customers { get; set; }
+    public DbSet<OrderItem> OrderItems { get; set; }
+    public DbSet<Employee> Employees { get; set; }
+    public DbSet<Table> Tables { get; set; }
+    public DbSet<MenuItem> MenuItems { get; set; }
+    public DbSet<Restaurant> Restaurants { get; set; }
+    public DbSet<Order> Orders { get; set; }
+    public DbSet<Reservation> Reservations { get; set; }
+    public DbSet<ReservationWithCustomerAndRestaurant> ReservationWithCustomerAndRestaurant { get; set; }
+    public DbSet<EmployeeWithRestaurant> EmployeeWithRestaurant { get; set; }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    {
+        optionsBuilder.UseSqlServer(Configuration.GetConnectionString());
+    }
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        MapDatabaseViews(modelBuilder);
+        MapDatabaseFunctions(modelBuilder);
+        MapRelationshipsAndForeignKeys(modelBuilder);
+        SeedData(modelBuilder);
+    }
+
+    private void MapDatabaseFunctions(ModelBuilder modelBuilder)
+    {
+        modelBuilder.HasDbFunction(typeof(RestaurantReservationDbContext)
+                    .GetMethod(nameof(GetTotalRevenue), new[] { typeof(int) }))
+                    .HasName("GetTotalRevenue");
+    }
+
+    public decimal GetTotalRevenue(int restaurantId)
+        => throw new NotSupportedException("This function is for use with EF Core only and cannot be called directly.");
+
+    private static void MapDatabaseViews(ModelBuilder modelBuilder)
```

Comment:
NIT: it is better to have a specific config class for each table containing its configuration alongside any other specific data validations (instead of using data annotations on the db model)